rands.c

```diff
@@ -19,38 +20,30 @@ long _randseed = 0L;
 
 /*****************************************************/
 /**
-  \fn void RandSeed(signed long seed)
-  \brief Resets the random number seed.
+  \brief Sets the random number seed. This should only be done ONCE per iteration.
 
- The seed is set to negative when this routine
- is called, so the generator routines
- ( eg, RandUni()) can tell that it has
- changed.  If called with seed==0,
- _randseed is reset from process time.
- '% 0xffff' is due to a bug in RandUni()
- that conks if seed is too large; should
- be removed in the near future.
+  \param Seed is the initial state of the system. 0 indicates a random seed.
+  \param pcg_rng is the random number generator to set.
 
- \param seed The seed.
-
- cwb - 6/27/00
 */
-void RandSeed(signed long seed) {
+void RandSeed(signed long seed, pcg32_random_t* pcg_rng) {

```

Comment:
The body of this function must have a `#ifdef RSOILWAT` case -- c code compiled for the R package `rSOILWAT2` must use the R API RNGs (see https://cran.r-project.org/doc/manuals/R-exts.html#Writing-portable-packages and https://cran.r-project.org/doc/manuals/R-exts.html#Random-numbers).